lib/poller.js

```diff
@@ -281,7 +281,15 @@ class Poller {
     // Testing data property
     if (namingConvention && externalData) {
       externalData.forEach((secretProperty, index) => {
-        if (!reNaming.test(secretProperty.key)) {
+        if (secretProperty.path) {
+          if (!reNaming.test(secretProperty.path)) {
+            allowed = false
+            reason = `path ${secretProperty.path} does not match naming convention ${namingConvention}`
+            return {
+              allowed, reason
+            }
+          }
+        } else if (!reNaming.test(secretProperty.key)) {
```

Comment:
While this shouldn't happen due to both CRD validation, and then later the backend implementation shouldn't use both path and key. It feels better to have them both validated and not just path if they for some reason both exist (like in the unit test)

```suggestion
        }
        if (!reNaming.test(secretProperty.key)) {
```

Reply:
Done.